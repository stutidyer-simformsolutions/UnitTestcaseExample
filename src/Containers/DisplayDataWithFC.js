/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from "react";
import {
  Button,
  FlatList,
  SafeAreaView,
  StyleSheet,
  Text,
  TextInput,
  View
} from "react-native";

const Section = ({ title, buttonTitle }) => {
  const [data, setData] = React.useState(null);
  const [myArray, setMyArray] = React.useState([]);
  const [isEditPressed, setIsEditPressed] = React.useState(false);
  const [currentIndex, setCurrentIndex] = React.useState(0);
  const onEditPress = (value, index) => {
    setData(value);
    setCurrentIndex(index);
    setIsEditPressed(true);
  };

  const onDeletePress = (index) => {
    setCurrentIndex(index);
    const arr = myArray;
    arr.splice(index, 1);
    setMyArray([...arr]);
  };

  onButtonPress = () => {
    if (isEditPressed) {
      const arr = myArray;
      arr.splice(currentIndex, 1, data);
      setMyArray(arr);
      setIsEditPressed(false);
    } else {
      setMyArray((oldArray) => [...oldArray, data]);
    }
    setData("");
  };
  return (
    <View style={styles.sectionContainer}>
      <Text style={styles.sectionTitle}>{title}</Text>
      <TextInput
        style={styles.input}
        onChangeText={setData}
        value={data}
        placeholder="Please enter the number"
        keyboardType="numeric"
        testID="section-textInput"
      />

      <Button
        disabled={!data ? true : false}
        title={buttonTitle}
        testID="section-button"
        onPress={onButtonPress}
      />
      <DisplayData
        data={myArray}
        onEditPress={onEditPress}
        onDeletePress={onDeletePress}
      />
    </View>
  );
};

const DisplayData = (props) => {
  const { onEditPress, onDeletePress, data } = props;
  const Item = ({ title, index }) => (
    <View style={styles.item}>
      <Text style={styles.title}>{title}</Text>
      <Button
        title={"Edit"}
        onPress={() => onEditPress(title, index)}
        testID={"edit-button"}
      />
      <Button
        title={"Delete"}
        onPress={() => onDeletePress(index)}
        testID={"delete-button"+ index}
      />
    </View>
  );

  const renderItem = ({ item, index }) => {
    return <Item title={item} index={index} />;
  };

  return (
    <FlatList
      data={data}
      testID={'data-flatlist'}
      renderItem={renderItem}
      keyExtractor={(item) => item}
    />
  );
};

const DisplayDataWithFC = () => {
  return (
    <SafeAreaView
      style={styles.mainContainerStyle}
    >
      <Section title={"Enter Value"} buttonTitle={"Press me"} />
    </SafeAreaView>
  );
};

export const styles = StyleSheet.create({
  mainContainerStyle: { flex: 1, marginHorizontal: 25 },
  sectionContainer: {
    marginTop: 32,
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: "600",
  },
  item: {
    flexDirection: "row",
    justifyContent: "space-between",
    padding: 5,
  },
  title: {
    fontSize: 17,
    top: 10,
  },
  input: {
    padding: 10,
    height: 40,
    marginVertical: 12,
    borderWidth: 1,
  },
});

export default DisplayDataWithFC;
