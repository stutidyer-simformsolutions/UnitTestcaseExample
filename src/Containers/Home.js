import React from "react";
import { Button, StyleSheet, Text, View } from "react-native";

const Home = (props) => {
  return (
    <View style={styles.container}>
      <Text
        style={styles.textStyle}
      >{`Hello Everyone! \n Welcome to Unit Testcase Demo!`}</Text>
      <View style={styles.buttonContainer}>
        <Button
          title="Demo with Statefull Component"
          onPress={() => props.navigation.navigate("DisplayDataWithClass")}
          style={styles.buttonStyle}
          testID="class-component-button"
        />
        <Button
          title="Demo with Stateless Component"
          onPress={() => props.navigation.navigate("DisplayDataWithFC")}
          style={styles.buttonStyle}
          accessibilityLabel="Press Me"
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  buttonContainer: { flex: 0.8, marginVertical: 25, justifyContent: "center" },
  textStyle: {
    textAlign: "center",
    marginVertical: 8,
    fontSize: 25,
    lineHeight: 50,
  },
  buttonStyle: {
    // height: 120,
    // width: 180
  },
});

export default Home;
