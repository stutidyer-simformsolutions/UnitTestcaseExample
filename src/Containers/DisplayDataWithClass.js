import React, { Component } from "react";
import {
  Button,
  FlatList,
  SafeAreaView,
  Text,
  TextInput,
  View,
} from "react-native";
import { styles } from "./DisplayDataWithFC";

export default class DisplayDataWithClass extends Component {
  constructor(props) {
    super(props);

    this.state = {
      data: null,
      myArray: [],
      isEditPressed: false,
      currentIndex: 0,
    };
  }

  onButtonPress = () => {
    const { isEditPressed, myArray, data, currentIndex } = this.state;
    if (isEditPressed) {
      const arr = myArray;
      arr.splice(currentIndex, 1, data);
      this.setState({ myArray: arr, isEditPressed: false });
    } else {
      this.setState((prevState) => ({
        myArray: [...this.state.myArray, this.state.data],
      }));
    }
    this.setState({ data: "" });
  };

  onChangeText = (text) => {
    this.setState({ data: text });
  };

  onEditPress = (item, index) => {
    this.setState({ data: item, currentIndex: index, isEditPressed: true });
  };
  onDeletePress = (index) => {
    const arr = this.state.myArray;
    arr.splice(index, 1);
    this.setState({ myArray: arr });
  };

  renderItem = ({ item, index }) => {
    return (
      <View style={styles.item}>
        <Text style={styles.title}>{item}</Text>
        <Button
          title={"Edit"}
          onPress={() => this.onEditPress(item, index)}
          testID={"edit-button"}
        />
        <Button
          title={"Delete"}
          onPress={() => this.onDeletePress(index)}
          testID={"delete-button"}
        />
      </View>
    );
  };

  renderData = () => {
    const { myArray } = this.state;
    return (
      <FlatList
        contentContainerStyle={{ flexGrow: 1 }}
        data={myArray}
        renderItem={this.renderItem}
        keyExtractor={(item) => item}
      />
    );
  };

  renderContent = () => {
    const { data } = this.state;
    return (
      <View style={styles.sectionContainer}>
        <Text style={styles.sectionTitle}>{"Enter Value"}</Text>
        <TextInput
          style={styles.input}
          onChangeText={this.onChangeText}
          value={data}
          placeholder="Please enter the number"
          keyboardType="numeric"
          testID="section-textInput"
        />

        <Button
          disabled={!data ? true : false}
          title={"Press me"}
          testID="section-button"
          onPress={this.onButtonPress}
        />
        {this.renderData()}
      </View>
    );
  };

  render() {
    return (
      <SafeAreaView style={styles.mainContainerStyle}>
        {this.renderContent()}
      </SafeAreaView>
    );
  }
}
