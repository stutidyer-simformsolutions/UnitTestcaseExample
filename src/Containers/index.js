import Home from "./Home";
import DisplayDataWithClass from "./DisplayDataWithClass";
import DisplayDataWithFC from "./DisplayDataWithFC";

export {DisplayDataWithClass, DisplayDataWithFC, Home};