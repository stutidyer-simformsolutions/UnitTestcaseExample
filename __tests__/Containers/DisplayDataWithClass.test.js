import React from "react";
import { DisplayDataWithClass } from "../../src/Containers";
import { shallow, mount } from "enzyme";
import { render, fireEvent } from "@testing-library/react-native";
import { Button } from "react-native";
describe("Display data with class", () => {
  const component = shallow(<DisplayDataWithClass />);
  const instance = component.instance();
  test("snapshot of the displayData with class component with shallow", () => {
    expect(component).toMatchSnapshot();
  });

  test("snapshot of the displayData with class component with mount", () => {
    const component = mount(<DisplayDataWithClass />);
    expect(component).toMatchSnapshot();
  });

  test("onChange event", () => {
    instance.onChangeText(10);
    expect(instance.state.data).toEqual(10);
  });

  test("when user add number into list", () => {
    instance.setState({ myArray: [10, 15], currentIndex: 2, data: 16 }, () => {
      instance.onButtonPress();
    });
    expect(instance.state.myArray).toEqual([10, 15, 16]);
  });

  test("when user press edit button", () => {
    instance.setState(
      { myArray: [10, 15], currentIndex: 1, data: 15, isEditPressed: true },
      () => {
        instance.onEditPress(16, 1);
        instance.onButtonPress();
      }
    );
    expect(instance.state.myArray).toEqual([10, 16]);
  });

  test("when user delete value from the list", () => {
    instance.setState({ myArray: [10, 15], currentIndex: 1, data: 15 }, () => {
      instance.onDeletePress(1);
    });
    expect(instance.state.myArray).toEqual([10]);
  });

  describe("renderItem", () => {
    test("renderItem with edit button call", () => {
      const renderItem = instance.renderItem({ item: 10, index: 0 });
      const { getByTestId: renderItemTestId } = render(renderItem);
      const { getByTestId } = render(<DisplayDataWithClass />);
      const editButton = renderItemTestId("edit-button");
      const textInput = getByTestId("section-textInput");
      fireEvent.changeText(textInput, 12);
      fireEvent.press(editButton);
      expect(textInput.props.value).toEqual(12);
    });

    test("renderItem with delete button call", () => {
      instance.setState({ myArray: [10], data: 10 }, () => {
        const renderItem = instance.renderItem({ item: 10, index: 0 });
        const { getByTestId } = render(renderItem);
        const deleteButton = getByTestId("delete-button");
        fireEvent.press(deleteButton);
      });
      expect(instance.state.myArray).toEqual([]);
    });
  });

  test("keyExtractor of FlatList", () => {
    instance.setState({ myArray: [10, 15], data: 15 }, async () => {
      let keyExtractor = component.find("FlatList").at(0).getElement()
        .props.keyExtractor;
      keyExtractor();
      expect(await keyExtractor).toBeDefined();
    });
  });
});
