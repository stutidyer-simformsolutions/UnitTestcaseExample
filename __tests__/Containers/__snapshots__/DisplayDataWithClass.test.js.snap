// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display data with class snapshot of the displayData with class component with mount 1`] = `
<DisplayDataWithClass>
  <ForwardRef(SafeAreaView)
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 25,
      }
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "flex": 1,
          "marginHorizontal": 25,
        }
      }
    >
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Object {
            "flex": 1,
            "marginHorizontal": 25,
          }
        }
      >
        <View
          style={
            Object {
              "marginTop": 32,
            }
          }
        >
          <View
            style={
              Object {
                "marginTop": 32,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 24,
                  "fontWeight": "600",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 24,
                    "fontWeight": "600",
                  }
                }
              >
                Enter Value
              </Text>
            </Text>
            <TextInput
              keyboardType="numeric"
              onChangeText={[Function]}
              placeholder="Please enter the number"
              style={
                Object {
                  "borderWidth": 1,
                  "height": 40,
                  "marginVertical": 12,
                  "padding": 10,
                }
              }
              testID="section-textInput"
              value={null}
            >
              <TextInput
                keyboardType="numeric"
                onChangeText={[Function]}
                placeholder="Please enter the number"
                style={
                  Object {
                    "borderWidth": 1,
                    "height": 40,
                    "marginVertical": 12,
                    "padding": 10,
                  }
                }
                testID="section-textInput"
                value={null}
              />
            </TextInput>
            <Button
              disabled={true}
              onPress={[Function]}
              testID="section-button"
              title="Press me"
            >
              <TouchableOpacity
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": true,
                  }
                }
                disabled={true}
                onPress={[Function]}
                testID="section-button"
              >
                <TouchableOpacity
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "disabled": true,
                    }
                  }
                  disabled={true}
                  hostRef={null}
                  onPress={[Function]}
                  testID="section-button"
                >
                  <ForwardRef(AnimatedComponentWrapper)
                    accessibilityRole="button"
                    accessibilityState={
                      Object {
                        "disabled": true,
                      }
                    }
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        undefined,
                        Object {
                          "opacity": 1,
                        },
                      ]
                    }
                    testID="section-button"
                  >
                    <AnimatedComponent
                      accessibilityRole="button"
                      accessibilityState={
                        Object {
                          "disabled": true,
                        }
                      }
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          undefined,
                          Object {
                            "opacity": 1,
                          },
                        ]
                      }
                      testID="section-button"
                    >
                      <View
                        accessibilityRole="button"
                        accessibilityState={
                          Object {
                            "disabled": true,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                        testID="section-button"
                      >
                        <View
                          accessibilityRole="button"
                          accessibilityState={
                            Object {
                              "disabled": true,
                            }
                          }
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                          testID="section-button"
                        >
                          <View
                            style={
                              Array [
                                Object {},
                                Object {},
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {},
                                  Object {},
                                ]
                              }
                            >
                              <Text
                                disabled={true}
                                style={
                                  Array [
                                    Object {
                                      "color": "#007AFF",
                                      "fontSize": 18,
                                      "margin": 8,
                                      "textAlign": "center",
                                    },
                                    Object {
                                      "color": "#cdcdcd",
                                    },
                                  ]
                                }
                              >
                                <Text
                                  disabled={true}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#007AFF",
                                        "fontSize": 18,
                                        "margin": 8,
                                        "textAlign": "center",
                                      },
                                      Object {
                                        "color": "#cdcdcd",
                                      },
                                    ]
                                  }
                                >
                                  Press me
                                </Text>
                              </Text>
                            </View>
                          </View>
                          <PressabilityDebugView
                            color="cyan"
                          />
                        </View>
                      </View>
                    </AnimatedComponent>
                  </ForwardRef(AnimatedComponentWrapper)>
                </TouchableOpacity>
              </TouchableOpacity>
            </Button>
            <FlatList
              contentContainerStyle={
                Object {
                  "flexGrow": 1,
                }
              }
              data={Array []}
              keyExtractor={[Function]}
              renderItem={[Function]}
            >
              <VirtualizedList
                contentContainerStyle={
                  Object {
                    "flexGrow": 1,
                  }
                }
                data={Array []}
                getItem={[Function]}
                getItemCount={[Function]}
                keyExtractor={[Function]}
                removeClippedSubviews={false}
                renderItem={[Function]}
                viewabilityConfigCallbackPairs={Array []}
              >
                <VirtualizedListContextProvider
                  value={
                    Object {
                      "cellKey": null,
                      "debugInfo": Object {
                        "cellKey": "rootList",
                        "horizontal": false,
                        "listKey": "rootList",
                        "parent": undefined,
                      },
                      "getNestedChildState": [Function],
                      "getOutermostParentListRef": [Function],
                      "getScrollMetrics": [Function],
                      "horizontal": false,
                      "registerAsNestedChild": [Function],
                      "unregisterAsNestedChild": [Function],
                    }
                  }
                >
                  <ScrollView
                    contentContainerStyle={
                      Object {
                        "flexGrow": 1,
                      }
                    }
                    data={Array []}
                    getItem={[Function]}
                    getItemCount={[Function]}
                    keyExtractor={[Function]}
                    onContentSizeChange={[Function]}
                    onLayout={[Function]}
                    onMomentumScrollBegin={[Function]}
                    onMomentumScrollEnd={[Function]}
                    onScroll={[Function]}
                    onScrollBeginDrag={[Function]}
                    onScrollEndDrag={[Function]}
                    removeClippedSubviews={false}
                    renderItem={[Function]}
                    scrollEventThrottle={50}
                    stickyHeaderIndices={Array []}
                    viewabilityConfigCallbackPairs={Array []}
                  >
                    <RCTScrollView
                      contentContainerStyle={
                        Object {
                          "flexGrow": 1,
                        }
                      }
                      data={Array []}
                      getItem={[Function]}
                      getItemCount={[Function]}
                      keyExtractor={[Function]}
                      onContentSizeChange={[Function]}
                      onLayout={[Function]}
                      onMomentumScrollBegin={[Function]}
                      onMomentumScrollEnd={[Function]}
                      onScroll={[Function]}
                      onScrollBeginDrag={[Function]}
                      onScrollEndDrag={[Function]}
                      removeClippedSubviews={false}
                      renderItem={[Function]}
                      scrollEventThrottle={50}
                      stickyHeaderIndices={Array []}
                      viewabilityConfigCallbackPairs={Array []}
                    >
                      <RCTScrollView
                        contentContainerStyle={
                          Object {
                            "flexGrow": 1,
                          }
                        }
                        data={Array []}
                        getItem={[Function]}
                        getItemCount={[Function]}
                        keyExtractor={[Function]}
                        onContentSizeChange={[Function]}
                        onLayout={[Function]}
                        onMomentumScrollBegin={[Function]}
                        onMomentumScrollEnd={[Function]}
                        onScroll={[Function]}
                        onScrollBeginDrag={[Function]}
                        onScrollEndDrag={[Function]}
                        removeClippedSubviews={false}
                        renderItem={[Function]}
                        scrollEventThrottle={50}
                        stickyHeaderIndices={Array []}
                        viewabilityConfigCallbackPairs={Array []}
                      >
                        <View>
                          <View />
                        </View>
                      </RCTScrollView>
                    </RCTScrollView>
                  </ScrollView>
                </VirtualizedListContextProvider>
              </VirtualizedList>
            </FlatList>
          </View>
        </View>
      </RCTSafeAreaView>
    </RCTSafeAreaView>
  </ForwardRef(SafeAreaView)>
</DisplayDataWithClass>
`;

exports[`Display data with class snapshot of the displayData with class component with shallow 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "flex": 1,
      "marginHorizontal": 25,
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 32,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "600",
        }
      }
    >
      Enter Value
    </Text>
    <TextInput
      keyboardType="numeric"
      onChangeText={[Function]}
      placeholder="Please enter the number"
      style={
        Object {
          "borderWidth": 1,
          "height": 40,
          "marginVertical": 12,
          "padding": 10,
        }
      }
      testID="section-textInput"
      value={null}
    />
    <Button
      disabled={true}
      onPress={[Function]}
      testID="section-button"
      title="Press me"
    />
    <FlatList
      contentContainerStyle={
        Object {
          "flexGrow": 1,
        }
      }
      data={Array []}
      keyExtractor={[Function]}
      renderItem={[Function]}
    />
  </View>
</ForwardRef(SafeAreaView)>
`;
